### PR TITLE
Update seamless_cloning.cpp

### LINKED ISSUE #15927: could we include 4-channels images, too? i.e. 4D alpha images?
https://github.com/opencv/opencv/blob/1acadd363b0d0ffcdabac8af3196cb65bef426b1/modules/photo/src/seamless_cloning.cpp#L54
reason:  all images with 4 channels (alpha channels) are required to be converted to 3 channels on the client side, then back to 3 channels just for API compatibility purposes.

### PR DESCRIPTION
**Pull request by sturkmen72**
### This pullrequest changes

resolves #15927

### CODE REVIEW

File: modules/photo/src/seamless_cloning.cpp [hunk 1]
```diff
@@ -79,6 +79,12 @@ void cv::seamlessClone(InputArray _src, InputArray _dst, InputArray _mask, Point
 
     Rect roi_s = boundingRect(mask);
     if (roi_s.empty()) return;
+
+    if (p.x < 0)
+        p.x = dest.cols / 2;
+    if (p.y < 0)
+        p.y = dest.rows / 2;
+
     Rect roi_d(p.x - roi_s.width / 2, p.y - roi_s.height / 2, roi_s.width, roi_s.height);
```
Comment (alalek) [inline modules/photo/src/seamless_cloning.cpp]: Need to ensure that Rect args are still valid.

Comment (collinbrake) [reply modules/photo/src/seamless_cloning.cpp]: How can I pull sturkmen72's code in order to work on this?

**Comment by asmorkalov** [comment]
@alalek Do you have any remarks to the patch or I can squash and merge it?

**Comment by alalek** [comment]
@asmorkalov Please check issue's comments: #15928

**Comment by sturkmen72** [comment]
i reverted commits related with #15928

File: modules/photo/src/seamless_cloning.cpp [hunk 2]
```diff
@@ -51,8 +51,8 @@ static Mat checkMask(InputArray _mask, Size size)
 {
     Mat mask = _mask.getMat();
     Mat gray;
-    if (mask.channels() == 3)
-        cvtColor(mask, gray, COLOR_BGR2GRAY);
+    if (mask.channels() > 1)
+        cvtColor(mask, gray, COLOR_BGRA2GRAY);
```
Comment (alalek) [inline modules/photo/src/seamless_cloning.cpp]: > COLOR_BGR2GRAY

Should work with both BGR/BGRA input too.

**Pull request merged**